Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp

```diff
@@ -320,6 +454,36 @@ void webkit_user_content_manager_unregister_script_message_handler(WebKitUserCon
     manager->priv->userContentController->removeUserMessageHandlerForName(String::fromUTF8(name), API::ContentWorld::pageContentWorld());
 }
 
+/**
+ * webkit_user_content_manager_register_script_message_handler_with_reply:
+ * @manager: A #WebKitUserContentManager
+ * @name: Name of the script message channel
+ * @world_name: the name of a #WebKitScriptWorld
+ *
+ * Registers a new user script message handler in script world.
+ *
+ * Registers a new user script message handler in script world with name @world_name.
+ * If NULL is passed as the world_name, the default world will be used.
+ * See webkit_user_content_manager_register_script_message_handler() for full description.
+ *
+ * Registering a script message handler will fail if the requested
+ * name has been already registered before.
+ *
+ * Since: 2.38
+ */
+void webkit_user_content_manager_register_script_message_handler_with_reply(WebKitUserContentManager* manager, const char* name, const char* worldName)
+{
+    g_return_val_if_fail(WEBKIT_IS_USER_CONTENT_MANAGER(manager), FALSE);
+    g_return_val_if_fail(name, FALSE);
+
+    if (worldName == nullptr)
+        Ref<WebScriptMessageHandler> handler = WebScriptMessageHandler::create(makeUnique<ScriptMessageClientGtk>(manager, name, true), AtomString::fromUTF8(name), API::ContentWorld::pageContentWorld());
+    else
+        Ref <WebScriptMessageHandler> handler = WebScriptMessageHandler::create(makeUnique<ScriptMessageClientGtk>(manager, name, true), AtomString::fromUTF8(name), webkitContentWorld(worldName));
```

Comment:
This won't build. The code below uses the `handler` variable outside of the `if`-`else` statement, which means that it needs to be defined outside from it:

```cpp
    Ref<WebScriptMessageHandler> handler;
    if (worldName)
        handler = WebScriptMessageHandler::create(...);
    else
        handler = ...;

    return manager->priv->userContentController->addUserScriptMessageHandler(handler.get());